From 7ecc7b620bdb14bae2daecabb24a372b9052beb6 Mon Sep 17 00:00:00 2001
From: "haotian.qu" <haotian.qu@amlogic.com>
Date: Mon, 23 Oct 2023 19:31:12 +0800
Subject: [PATCH 3/4] SM1&G12B: cold start time optimization. [1/3]

PD#SWPL-139535

Problem:
Cold start time optimization.

Solution:
1. Improve emmc I/O performance.
2. Optimize startup process

Verify:
local PB

Signed-off-by: haotian.qu <haotian.qu@amlogic.com>
Change-Id: Iff7e6df45887b373c80b2935deba2b8f34e9ee48
---
 board/amlogic/configs/g12a_u200_debian_v1.h | 33 ++++++---------------
 board/amlogic/configs/g12b_w400_debian_v1.h | 33 ++++++---------------
 2 files changed, 18 insertions(+), 48 deletions(-)

diff --git a/board/amlogic/configs/g12a_u200_debian_v1.h b/board/amlogic/configs/g12a_u200_debian_v1.h
index c938d2e082f..9beda9d8bee 100644
--- a/board/amlogic/configs/g12a_u200_debian_v1.h
+++ b/board/amlogic/configs/g12a_u200_debian_v1.h
@@ -137,8 +137,6 @@
 #define CONFIG_EXTRA_ENV_SETTINGS \
 	"firstboot=1\0"\
 	"upgrade_step=0\0"\
-	"jtag=disable\0"\
-	"port_mode=0\0"\
 	"loadaddr=0x00020000\0"\
 	"dv_fw_addr=0xa00000\0"\
 	"panel_type=mipi_0\0" \
@@ -156,7 +154,6 @@
 	"fdtfile=sm1_s905d3_ac200_debian.dtb\0"\
 	"silent=1\0"\
 	"lcd_ctrl=0x00000000\0"\
-	"lcd_debug=0x00000000\0"\
 	"outputmode=1080p60hz\0"\
 	"vout_init=enable\0"\
 	"hdmimode=1080p60hz\0"\
@@ -203,14 +200,16 @@
 	"Irq_check_en=0\0"\
 	"reboot_mode_android=""normal""\0"\
 	"initargs="\
-		"rootflags=data=writeback rw rootfstype=ext4 console=tty0 console=ttyS0,115200 no_console_suspend " \
-		" earlycon=aml-uart,0xff803000 ramoops.pstore_en=1 ramoops.record_size=0x8000 ramoops.console_size=0x4000 "\
-		"scsi_mod.scan=async xhci_hcd.quirks=0x800000 gamma=0 "\
+		"rootflags=data=writeback rw rootfstype=ext4 loglevel=4 "\
+		"console=tty0 console=ttyS0,115200 no_console_suspend "\
+		"earlycon=aml-uart,0xff803000 ramoops.pstore_en=1 "\
+		"ramoops.record_size=0x8000 ramoops.console_size=0x4000 "\
+		"scsi_mod.scan=async xhci_hcd.quirks=0x800000 gamma=0 boot_source=emmc "\
 		"\0"\
 	"upgrade_check="\
 		"echo upgrade_step=${upgrade_step}; "\
 		"if itest ${upgrade_step} == 3; then "\
-			"run init_display; run storeargs; run update;"\
+			"run init_display;run update;"\
 		"else fi;"\
 		"\0"\
 	"storeargs="\
@@ -239,27 +238,21 @@
 		"get_rebootmode;"\
 		"if test ${reboot_mode} = factory_reset; then "\
 			"setenv reboot_mode_android ""normal"";"\
-			"run storeargs;"\
 			"run recovery_from_flash;"\
 		"else if test ${reboot_mode} = update; then "\
 			"setenv reboot_mode_android ""normal"";"\
-			"run storeargs;"\
 			"run update;"\
 		"else if test ${reboot_mode} = quiescent; then "\
 			"setenv reboot_mode_android ""quiescent"";"\
-			"run storeargs;"\
 			"setenv bootargs ${bootargs} androidboot.quiescent=1;"\
 		"else if test ${reboot_mode} = recovery_quiescent; then "\
 			"setenv reboot_mode_android ""quiescent"";"\
-			"run storeargs;"\
 			"setenv bootargs ${bootargs} androidboot.quiescent=1;"\
 			"run recovery_from_flash;"\
 		"else if test ${reboot_mode} = cold_boot; then "\
 			"setenv reboot_mode_android ""normal"";"\
-			"run storeargs;"\
 		"else if test ${reboot_mode} = fastboot; then "\
 			"setenv reboot_mode_android ""normal"";"\
-			"run storeargs;"\
 			"fastboot;"\
 		"fi;fi;fi;fi;fi;fi;"\
 		"\0"\
@@ -276,7 +269,7 @@
 	"factory_reset_poweroff_protect="\
 		"echo wipe_data=${wipe_data}; echo wipe_cache=${wipe_cache};"\
 		"if test ${wipe_data} = failed; then "\
-			"run init_display; run storeargs;"\
+			"run init_display;"\
 			"if mmcinfo; then "\
 				"run recovery_from_sdcard;"\
 			"fi;"\
@@ -286,7 +279,7 @@
 			"run recovery_from_flash;"\
 		"fi; "\
 		"if test ${wipe_cache} = failed; then "\
-			"run init_display; run storeargs;"\
+			"run init_display;"\
 			"if mmcinfo; then "\
 				"run recovery_from_sdcard;"\
 			"fi;"\
@@ -425,14 +418,10 @@
 		"boot_targets=spi usb0 mmc0 mmc1 storeboot pxe dhcp\0"
 
 #define CONFIG_PREBOOT  \
-			"run bcb_cmd; "\
-			"run factory_reset_poweroff_protect;"\
 			"run upgrade_check;"\
 			"run init_display;"\
 			"run storeargs;"\
 			"run upgrade_key;" \
-			"run port_mode_change;"\
-			"forceupdate;" \
 			"bcb uboot-command;"\
 			"run switch_bootmode;"
 
@@ -478,7 +467,7 @@
 #define CONFIG_SYS_BOOTM_LEN (64 << 20) /* Increase max gunzip size*/
 
 /* cpu */
-#define CONFIG_CPU_CLK					1200 //MHz. Range: 360-2000, should be multiple of 24
+#define CONFIG_CPU_CLK	1920 //MHz. Range: 360-2000, should be multiple of 24
 
 /* ATTENTION */
 /* DDR configs move to board/amlogic/[board]/firmware/timing.c */
@@ -517,10 +506,6 @@
  */
 /* axg only support slc nand */
 /* swither for mtd nand which is for slc only. */
-/* support for mtd */
-#define CONFIG_AML_MTD 1
-/* support for nftl */
-//#define CONFIG_AML_NAND	1
 
 #if defined(CONFIG_AML_NAND) && defined(CONFIG_AML_MTD)
 #error CONFIG_AML_NAND/CONFIG_AML_MTD can not support at the sametime;
diff --git a/board/amlogic/configs/g12b_w400_debian_v1.h b/board/amlogic/configs/g12b_w400_debian_v1.h
index d037a10d7c2..efa713142ef 100644
--- a/board/amlogic/configs/g12b_w400_debian_v1.h
+++ b/board/amlogic/configs/g12b_w400_debian_v1.h
@@ -119,8 +119,6 @@
 #define CONFIG_EXTRA_ENV_SETTINGS \
 		"firstboot=1\0"\
 		"upgrade_step=0\0"\
-		"jtag=disable\0"\
-		"port_mode=0\0"\
 		"loadaddr=0x00020000\0"\
 		"dv_fw_addr=0xa00000\0"\
 		"panel_type=mipi_0\0" \
@@ -138,7 +136,6 @@
 		"fdtfile=g12b_a311d_w400_debian.dtb\0" \
 		"silent=1\0"\
 		"lcd_ctrl=0x00000000\0" \
-		"lcd_debug=0x00000000\0" \
 		"outputmode=1080p60hz\0" \
 		"vout_init=enable\0" \
 		"hdmimode=1080p60hz\0" \
@@ -185,14 +182,16 @@
 		"Irq_check_en=0\0"\
 		"reboot_mode_android=""normal""\0"\
 		"initargs="\
-			"rootflags=data=writeback rw rootfstype=ext4 console=tty0 console=ttyS0,115200 no_console_suspend " \
-			" earlycon=aml-uart,0xff803000 ramoops.pstore_en=1 ramoops.record_size=0x8000 ramoops.console_size=0x4000 "\
-			"scsi_mod.scan=async xhci_hcd.quirks=0x800000 gamma=0 "\
+			"rootflags=data=writeback rw rootfstype=ext4 loglevel=4 " \
+			"console=tty0 console=ttyS0,115200 no_console_suspend " \
+			"earlycon=aml-uart,0xff803000 ramoops.pstore_en=1 " \
+			"ramoops.record_size=0x8000 ramoops.console_size=0x4000 "\
+			"scsi_mod.scan=async xhci_hcd.quirks=0x800000 gamma=0 boot_source=emmc "\
 			"\0"\
 		"upgrade_check="\
 			"echo upgrade_step=${upgrade_step}; "\
 			"if itest ${upgrade_step} == 3; then "\
-				"run init_display; run storeargs; run update;"\
+				"run init_display; run update;"\
 			"else fi;"\
 			"\0"\
 		"storeargs="\
@@ -220,27 +219,21 @@
 			"get_rebootmode;"\
 			"if test ${reboot_mode} = factory_reset; then "\
 					"setenv reboot_mode_android ""normal"";"\
-					"run storeargs;"\
 					"run recovery_from_flash;"\
 			"else if test ${reboot_mode} = update; then "\
 					"setenv reboot_mode_android ""normal"";"\
-					"run storeargs;"\
 					"run update;"\
 			"else if test ${reboot_mode} = quiescent; then "\
 					"setenv reboot_mode_android ""quiescent"";"\
-					"run storeargs;"\
 					"setenv bootargs ${bootargs} androidboot.quiescent=1;"\
 			"else if test ${reboot_mode} = recovery_quiescent; then "\
 					"setenv reboot_mode_android ""quiescent"";"\
-					"run storeargs;"\
 					"setenv bootargs ${bootargs} androidboot.quiescent=1;"\
 					"run recovery_from_flash;"\
 			"else if test ${reboot_mode} = cold_boot; then "\
 					"setenv reboot_mode_android ""normal"";"\
-					"run storeargs;"\
 			"else if test ${reboot_mode} = fastboot; then "\
 				"setenv reboot_mode_android ""normal"";"\
-				"run storeargs;"\
 				"fastboot;"\
 		    "fi;fi;fi;fi;fi;fi;"\
 			"\0" \
@@ -256,7 +249,7 @@
 		"factory_reset_poweroff_protect="\
 			"echo wipe_data=${wipe_data}; echo wipe_cache=${wipe_cache};"\
 			"if test ${wipe_data} = failed; then "\
-				"run init_display; run storeargs;"\
+				"run init_display;"\
 				"if mmcinfo; then "\
 					"run recovery_from_sdcard;"\
 				"fi;"\
@@ -266,7 +259,7 @@
 				"run recovery_from_flash;"\
 			"fi; "\
 			"if test ${wipe_cache} = failed; then "\
-				"run init_display; run storeargs;"\
+				"run init_display;"\
 				"if mmcinfo; then "\
 					"run recovery_from_sdcard;"\
 				"fi;"\
@@ -405,14 +398,10 @@
 		"boot_targets=spi usb0 mmc0 mmc1 storeboot pxe dhcp\0"
 
 #define CONFIG_PREBOOT  \
-			"run bcb_cmd; "\
-			"run factory_reset_poweroff_protect;"\
 			"run upgrade_check;"\
 			"run init_display;"\
 			"run storeargs;"\
 			"run upgrade_key;"\
-			"run port_mode_change;"\
-			"forceupdate;" \
 			"bcb uboot-command;"\
 			"run switch_bootmode;"
 
@@ -458,7 +447,7 @@
 #define CONFIG_SYS_BOOTM_LEN (64 << 20) /* Increase max gunzip size*/
 
 /* cpu */
-#define CONFIG_CPU_CLK					1200 //MHz. Range: 360-2000, should be multiple of 24
+#define CONFIG_CPU_CLK 1200 //MHz. Range: 360-2000, should be multiple of 24
 
 /* ATTENTION */
 /* DDR configs move to board/amlogic/[board]/firmware/timing.c */
@@ -497,10 +486,6 @@
 */
 /* axg only support slc nand */
 /* swither for mtd nand which is for slc only. */
-/* support for mtd */
-#define CONFIG_AML_MTD 1
-/* support for nftl */
-//#define CONFIG_AML_NAND	1
 
 #if defined(CONFIG_AML_NAND) && defined(CONFIG_AML_MTD)
 #error CONFIG_AML_NAND/CONFIG_AML_MTD can not support at the sametime;
-- 
2.25.1

